src/internal/proc.js

```diff
@@ -481,7 +485,17 @@ export default function proc(
 
     try {
       suspend()
-      const task = proc(taskIterator, subscribe, dispatch, getState, options, effectId, fn.name, (detached ? null : noop))
+      const task = proc(
+        taskIterator,
+        subscribe,
+        dispatch,
+        getState,
+        (detached ? null : taskContext), // TODO: topContext should be passed in case of spawn?
```

Comment:
reasoning behind it - `spawn` creates a `detached` task, which is acting like a top task, so would it be good to propagate `taskContext` (parent context) to it? behaviour would differ from other propagations (error/cancellation) in terms of detached (`spawn`) and non-detached tasks (`fork`)